src/ui/state_manager.hh

```diff
@@ -0,0 +1,25 @@
+#ifndef STATE_MANAGER_HH
+#define STATE_MANAGER_HH
+
+#include <config.hh>
+#include <QMutex>
+
+class StateManager: public QObject
+{
+  Q_OBJECT
+
+public:
+  StateManager( QWidget * parent, Config::Class & cfg );
+
+  void setState( QByteArray state, QByteArray geometry );
+
+private:
+  bool _dirty;
+  Config::Class & _cfg;
```

Comment:
Maybe we should store window states in a different file.

Serializing the entire cfg to XML is a relatively costly procedure.

The original GD's resaving config at exit is a symptom of bad programming. For user-decided settings, the config shouldn't be changed at all after closing the Preferences & Edit Dictionary dialogs. In contrast, preserving UI state/geometry and other runtime states is more machine-centric.

Reply:
> Maybe we should store window states in a different file.

sounds nice

Reply:
> Maybe we should store window states in a different file.

sounds nice

Reply:
what about qsettings

Reply:
By default, `QSettings` uses different strategies for different platforms and the path of setting file depends on organization+app name.

If you want to use it, I think we should avoid these differences (aka pointless complexity) by just using the `QSettings::IniFormat` for all platforms, and set the path to `${GD's config path}/stats.ini` (for example, by using this constructor https://doc.qt.io/qt-6/qsettiQngs.html#setPath).